assets/src/stories-editor/components/higher-order/with-replaced-full-size-image.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { dispatch, select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { MAX_IMAGE_SIZE_SLUG } from '../../constants';
+
+const BLOCK_EDITOR_STORE = 'core/block-editor';
+
+export default createHigherOrderComponent(
+	( BlockEdit ) => {
+		return class extends BlockEdit {
+			/**
+			 * The class constructor.
+			 */
+			constructor() {
+				super( ...arguments );
+				this.componentDidUpdate = this.componentDidUpdate.bind( this );
+				this.ampReplaceFullSizeImage = this.ampReplaceFullSizeImage.bind( this );
+			}
+
+			/**
+			 * When the component updates, if it's for the Image block, replace the full size image.
+			 */
+			componentDidUpdate() {
+				const { name } = this.props;
+				if ( 'core/image' === name ) {
+					this.ampReplaceFullSizeImage();
+				}
+
+				if ( super.componentDidUpdate ) {
+					super.componentDidUpdate();
```

Comment:
I'm no React expert, but shouldn't this be passing the args passed into `componentDidUpdate`? In other words:

```js
super.componentDidUpdate( ...arguments );
```

Reply:
Ah, good point.